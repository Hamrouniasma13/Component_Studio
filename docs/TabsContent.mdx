---
name: Tabs content (deprecated)
route: /tabs-content
menu: 🗑️ Deprecated
---

import { Playground } from 'docz'
import { TabsContent } from '../src'

# Help Content

<Playground>
  <TabsContent
    title="Title"
    defaultIndex={0}
    items={[
      {
        title: 'First title',
        content:
          'Sint dolor deserunt minim mollit veniam reprehenderit deserunt eiusmod magna commodo. Nisi minim ad dolore incididunt ut consectetur enim ipsum dolor excepteur. Elit culpa ad non ipsum. Dolore anim pariatur duis voluptate.',
      },
      {
        title: 'Second title',
        content:
          'Ut ut irure ipsum velit minim ex commodo. Aliqua aute quis magna exercitation voluptate. Ullamco est anim consequat ad aute.',
      },
      {
        title: 'Third title',
        content:
          'Elit excepteur ad eiusmod ipsum voluptate ex. Qui Lorem enim eu elit velit eiusmod incididunt occaecat. Tempor nostrud amet occaecat ut ex nulla do occaecat reprehenderit ipsum nostrud dolor consectetur.',
      },
    ]}
  />
</Playground>
