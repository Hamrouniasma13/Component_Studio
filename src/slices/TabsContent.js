/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import React from 'react'
import PropTypes from 'prop-types'
import { useMedia } from 'react-use'

import Slice from './Slice'
import Chevron from '../components/AnimatedIcons/Chevron'
import Text from '../components/Text'

function TabElement({ title, content, onSelect, active }) {
  const theme = React.useContext(ThemeContext)

  const styles = {
    titleWrapper: {
      position: 'relative',
      label: 'TabItemTitle',
      userSelect: 'none',
      width: '100%',
      textTransform: 'uppercase',
      color: theme.colors.primary,
      display: 'table',
      borderBottom: '1px solid #e5e5e5',
      height: '61px',
      textDecoration: 'none',
      [theme.media.tablet]: {
        color: theme.colors.black,
        paddingLeft: '0',
        paddingRight: '0',
        paddingTop: '0',
        display: 'block',
        textTransform: 'none',
        textDecoration: active ? 'underline' : 'none',
        fontWeight: active ? '700' : '300',
        height: 'auto',
        borderBottom: 0,
        paddingBottom: '30px'
      }
    },
    title: {
      display: 'table-cell',
      verticalAlign: 'middle',
      paddingLeft: '15px',
      paddingRight: '15px',
      fontSize: '14px',
      fontWeight: '700',
      cursor: 'pointer'
    },
    chevron: {
      color: theme.colors.primary,
      position: 'absolute',
      top: '30px',
      right: '0',
      marginTop: -8,
      [theme.media.tablet]: {
        display: 'none !important'
      }
    },
    content: {
      margin: '20px auto',
      display: active ? 'block' : 'none',
      [theme.media.tablet]: {
        display: 'none'
      }
    }
  }

  return (
    <div>
      <div css={styles.titleWrapper} onClick={onSelect}>
        <div css={styles.title}>{title}</div>
        <Chevron css={styles.chevron} size={16} open={active} />
      </div>
      <div css={styles.content}>{content}</div>
    </div>
  )
}

function TabContent({ title, content }) {
  const theme = React.useContext(ThemeContext)

  const styles = {
    wrapper: {
      display: 'none',
      [theme.media.tablet]: {
        display: 'block',
        width: '80%',
        float: 'right',
        background: 'white'
      }
    },
    innerWrapper: {
      padding: 20,
      border: `1px solid ${theme.colors.lightGray}`,
      boxShadow: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
      borderRadius: 3
    }
  }

  return (
    <div css={styles.wrapper}>
      <div css={styles.innerWrapper}>
        <Text.h3>{title}</Text.h3>
        <div css={{ marginTop: 20 }}>{content}</div>
      </div>
    </div>
  )
}

export default function TabsContent({ title, items, defaultIndex }) {
  const [currentItem, setCurrentItem] = React.useState(defaultIndex)
  const theme = React.useContext(ThemeContext)
  const aboveTablet = useMedia(`(min-width : ${theme.breakpoints.tablet}px)`)

  const styles = {
    list: {
      [theme.media.tablet]: {
        marginTop: 15,
        width: '20%',
        float: 'left'
      }
    }
  }

  return (
    <Slice title={title}>
      <div css={styles.list}>
        {items &&
          items.map((tab, key) => (
            <TabElement
              {...tab}
              key={key}
              active={key === currentItem}
              onSelect={() =>
                setCurrentItem(
                  key === currentItem ? (aboveTablet ? key : -1) : key
                )
              }
            />
          ))}
      </div>

      <TabContent {...items[Math.max(currentItem, 0)]} />
    </Slice>
  )
}

TabsContent.propTypes = {
  title: Slice.propTypes.title,
  items: PropTypes.arrayOf(PropTypes.object),
  defaultIndex: PropTypes.number
}
