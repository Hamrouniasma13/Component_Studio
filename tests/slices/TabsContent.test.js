import React from 'react';
import renderer from 'react-test-renderer';

import TabsContent from '../../slices/TabsContent';

describe('Component: <TabsContent />', () => {
  it('snapshot', () => {
    const tree = renderer.create(
      <TabsContent
        items={[
          {
            title: "Se connecter Ã  l'espace client",
            content: '1Ex eiusmod id consequat laborum.'
          },
          {
            title: 'Mon identifiant de connexion',
            content:
              '2Reprehenderit irure eiusmod duis sint officia proident elit.'
          },
          {
            title: 'Code confidentielle',
            content:
              '3Laboris labore nisi nostrud enim commodo ad consequat sit ullamco ex velit occaecat pariatur.'
          }
        ]}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
