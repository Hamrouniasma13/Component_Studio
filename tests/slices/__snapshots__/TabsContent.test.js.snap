// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TabsContent /> snapshot 1`] = `
.emotion-15 {
  background-color: #fafafa;
}

.emotion-14 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
}

@media only screen and (min-width :768px) {
  .emotion-14 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-14 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-14 {
    width: 1140px;
  }
}

.emotion-14:after {
  display: block;
  content: "";
  clear: both;
}

@media only screen and (min-width :768px) {
  .emotion-14 {
    padding-top: 50px;
    padding-bottom: 50px;
  }
}

@media only screen and (min-width :768px) {
  .emotion-12 {
    margin-top: 30px;
    width: 20%;
    float: left;
  }
}

.emotion-2 {
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  text-transform: uppercase;
  color: #494df4;
  display: table;
  border-bottom: 1px solid #e5e5e5;
  height: 61px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media only screen and (min-width :768px) {
  .emotion-2 {
    color: #000000;
    padding-left: 0px;
    padding-right: 0px;
    padding-top: 0px;
    display: block;
    text-transform: none;
    -webkit-text-decoration: none;
    text-decoration: none;
    font-weight: 300;
    height: auto;
    border-bottom: 0;
    padding-bottom: 30px;
  }
}

.emotion-0 {
  display: table-cell;
  vertical-align: middle;
  padding-left: 15px;
  padding-right: 15px;
  font-size: 14px;
  font-weight: 700;
}

.emotion-1 {
  color: inherit;
  height: 28px;
  width: 28px;
  -webkit-transition: -webkit-transform 0.3s,color 0.3s;
  -webkit-transition: transform 0.3s,color 0.3s;
  transition: transform 0.3s,color 0.3s;
  -webkit-transform: translate(0,-50%) rotate(180deg);
  -ms-transform: translate(0,-50%) rotate(180deg);
  transform: translate(0,-50%) rotate(180deg);
  color: #494df4;
  position: absolute;
  top: 30px;
  right: 0px;
}

@media only screen and (min-width :768px) {
  .emotion-1 {
    display: none !important;
  }
}

.emotion-3 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
  max-width: 798px;
  margin-bottom: 50px;
  font-size: 16px;
  line-height: 1.5;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  color: #333333;
  margin: 20px auto;
  display: none;
}

@media only screen and (min-width :768px) {
  .emotion-3 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-3 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-3 {
    width: 1140px;
  }
}

@media only screen and (min-width :768px) {
  .emotion-3 {
    font-size: 18px;
    line-height: 1.67;
  }
}

@media only screen and (min-width :768px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-13 {
  display: none;
}

@media only screen and (min-width :768px) {
  .emotion-13 {
    display: block;
    width: 80%;
    float: right;
    background: white;
  }
}

<div
  className="emotion-15"
>
  <div
    className="emotion-14"
  >
    <div
      className="emotion-12"
    >
      <div>
        <div
          className="emotion-2"
          onClick={[Function]}
        >
          <div
            className="emotion-0"
          >
            Se connecter Ã  l'espace client
          </div>
          <div
            className="emotion-1"
            colorindex={undefined}
            fill="currentColor"
            open={false}
            size={28}
            style={
              Object {
                "alignItems": "center",
                "display": "inline-flex",
                "justifyContent": "center",
              }
            }
          >
            <svg
              fill="currentColor"
              height={28}
              style={
                Object {
                  "display": "inline-block",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 24 24"
              width={28}
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="emotion-3"
        >
          1Ex eiusmod id consequat laborum.
        </div>
      </div>
      <div>
        <div
          className="emotion-2"
          onClick={[Function]}
        >
          <div
            className="emotion-0"
          >
            Mon identifiant de connexion
          </div>
          <div
            className="emotion-1"
            colorindex={undefined}
            fill="currentColor"
            open={false}
            size={28}
            style={
              Object {
                "alignItems": "center",
                "display": "inline-flex",
                "justifyContent": "center",
              }
            }
          >
            <svg
              fill="currentColor"
              height={28}
              style={
                Object {
                  "display": "inline-block",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 24 24"
              width={28}
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="emotion-3"
        >
          2Reprehenderit irure eiusmod duis sint officia proident elit.
        </div>
      </div>
      <div>
        <div
          className="emotion-2"
          onClick={[Function]}
        >
          <div
            className="emotion-0"
          >
            Code confidentielle
          </div>
          <div
            className="emotion-1"
            colorindex={undefined}
            fill="currentColor"
            open={false}
            size={28}
            style={
              Object {
                "alignItems": "center",
                "display": "inline-flex",
                "justifyContent": "center",
              }
            }
          >
            <svg
              fill="currentColor"
              height={28}
              style={
                Object {
                  "display": "inline-block",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 24 24"
              width={28}
            >
              <path
                d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </div>
        </div>
        <div
          className="emotion-3"
        >
          3Laboris labore nisi nostrud enim commodo ad consequat sit ullamco ex velit occaecat pariatur.
        </div>
      </div>
    </div>
    <div
      className="emotion-13"
    />
  </div>
</div>
`;
